pycbc/events/status.py

```diff
@@ -0,0 +1,252 @@
+    start_pad = numpy.array(veto_table.getColumnByName('start_pad'))
+    end_pad = numpy.array(veto_table.getColumnByName('end_pad'))
+    
+    data = {}
+    for i in range(len(veto_table)):
+        if ifo[i] not in data:
+            data[ifo[i]] = {}
+        
+        if category[i] not in data[ifo[i]]:
+            data[ifo[i]][category[i]] = []
+            
+        veto_info = {'name': name[i],
+                     'version': version[i],
+                     'start': start[i],
+                     'end': end[i],
+                     'start_pad': start_pad[i],
+                     'end_pad': end_pad[i],
+                     }
+        data[ifo[i]][category[i]].append(veto_info)
+    return data
+
+def query_flag(ifo, segment_name, start_time, end_time,
+               source='any', server="segments.ligo.org",
+               veto_definer=None):
+    """Return the times where the flag is active 
+
+    Parameters
+    ----------
+    ifo: string
+        The interferometer to query (H1, L1).
+    segment_name: string
+        The status flag to query from LOSC.
+    start_time: int 
+        The starting gps time to begin querying from LOSC
+    end_time: int 
+        The end gps time of the query
+    source: str, Optional
+        Choice between "GWOSC" or "dqsegdb". If dqsegdb, the server option may
+        also be given. The default is to try GWOSC first then try dqsegdb.
+    server: str, Optional
+        The server path. Only used with dqsegdb atm.
+    veto_definer: str, Optional
+        The path to a veto definer to define groups of flags which
+        themselves define a set of segments.
+
+    Returns
+    ---------
+    segments: glue.segments.segmentlist
+        List of segments
+    """
+    if source == 'GWOSC' or source == 'any':
+        ### Special cases as the LOSC convention is backwards from normal
+        ### LIGO / Virgo operation!!!!
+        if (('_HW_INJ' in segment_name and not 'NO' in segment_name) or 
+           'VETO' in segment_name):
+            data = query_flag(ifo, 'DATA', start_time, end_time)
+             
+            if '_HW_INJ' in segment_name:
+                name = 'NO_' + segment_name    
+            else:
+                name = segment_name.replace('_VETO', '')
+
+            negate = query_flag(ifo, name, start_time, end_time)
+            return (data - negate).coalesce()
+
+        duration = end_time - start_time
+        url = 'https://www.gw-openscience.org/timeline/segments/json/O1/{}_{}/{}/{}/'
+        url = url.format(ifo, segment_name, start_time, duration)
+        
+        try:
+            flag_segments = json.loads(urllib.urlopen(url).read())['segments']
+        except:
+            msg = "Unable to find segments in GWOSC, check flag name or times"
+            if source == 'any':
+                return query_flag(ifo, segment_name, start_time, end_time,
+                                  source=='dqsegdb')
```

Comment:
Ah, I see, you'd fall through here ... But you dropped a bunch of important key-word arguments ... Somehow I've fallen through with source=False ... Oh I see, you used `==`, DOH!

... I'm editing some of this locally now.

Reply:
Uh... please don't edit locally that will conflict with me.